resources/views/layouts/partials/header/autocomplete.blade.php

```diff
@@ -1,56 +1,70 @@
 
-<label for="autocomplete-input" class="sr-only">@lang('Search')</label>
-<input
-    id="autocomplete-input"
-    placeholder="@lang('Search')"
-    class="{{ $inputClasses }}"
-    v-on:focus="$root.loadAutocomplete = true"
-    v-if="!$root.loadAutocomplete"
->
+<div v-if="!$root.loadAutocomplete" class="relative w-full">
+    <label for="autocomplete-input" class="sr-only">@lang('Search')</label>
+    <input
+        id="autocomplete-input"
+        placeholder="@lang('What are you looking for?')"
+        class="{{ $inputClasses }}"
+        v-on:focus="$root.loadAutocomplete = true"
+    >
+    <x-rapidez::autocomplete.magnifying-glass />
+</div>
 
 <autocomplete
-    v-if="$root.loadAutocomplete"
+    v-else
     v-on:mounted="() => window.document.getElementById('autocomplete-input').focus()"
     v-bind:additionals="{{ json_encode(config('rapidez.frontend.autocomplete.additionals')) }}"
     v-bind:debounce="{{ config('rapidez.frontend.autocomplete.debounce') }}"
     v-bind:size="{{ config('rapidez.frontend.autocomplete.size') }}"
-    class="w-full"
+    class="relative w-full"
     v-cloak
 >
-    <x-rapidez::reactive-base slot-scope="{ results, resultsCount, searchAdditionals, debounce, size, highlight }">
+    <x-rapidez::reactive-base slot-scope="autocompleteScope">
+        <div
+            class="z-header-autocomplete-overlay pointer-events-none fixed inset-0 cursor-pointer bg-black/40 opacity-0 transition duration-500"
+            :class="autocompleteScope.overlay ? 'pointer-events-auto opacity-100 prevent-scroll' : 'opacity-0 pointer-events-none '"
+        ></div>
+        <x-rapidez::autocomplete.magnifying-glass v-bind:class="{ 'bg-primary text-white': autocompleteScope.overlay }" />
         <data-search
-            placeholder="@lang('Search')"
+            placeholder="@lang('What are you looking for?')"
             v-on:value-selected="search"
             component-id="autocomplete"
             :inner-class="{ input: '{{ $inputClasses }}' }"
-            {{-- These classes are only used when you come from a page with a product listing, --}}
-            {{-- click on a link that leads to a 404 page and try to use the search there --}}
-            class="relative [&_*]:!m-0 [&_[isclearicon=]]:!mr-2 [&_.cancel-icon]:!fill-[#595959] [&_[groupposition=right]]:!absolute [&_[groupposition=right]]:!top-2/4 [&_[groupposition=right]]:!right-0 [&_[groupposition=right]]:!-translate-y-2/4"
+            class="relative [&_*]:!m-0"
             :data-field="Object.keys(config.searchable)"
             :field-weights="Object.values(config.searchable)"
             :show-icon="false"
             fuzziness="AUTO"
-            :debounce="debounce"
-            :size="size"
+            :debounce="autocompleteScope.debounce"
+            v-on:blur="autocompleteScope.showOverlay(false)"
+            v-on:focus="autocompleteScope.showOverlay(true)"
+            :size="autocompleteScope.size"
```

Comment:
While we're here, can we improve the consistency of using `v-bind:` everywhere instead of just `:`?